### PR TITLE
Fix 854

### LINKED ISSUE #854: climb:text-style-to-font there is no applicable method for ps/pdf/raster-image backends
For the output only backends ps pdf and raster-image there isn't applicabe  method for the generic function `climb:text-style-to-font` that is called from generic function `climb:text-style-character-width`. Maybe it is not the only missing method. I have found the error trying to print a scigraph plot to a pdf file.

### PR DESCRIPTION
**Pull request by admich**
In the first commit I remove cimb:text-style-to-font because it is a replica of clim:text-style-mapping.

In the second commit I implement a method for climb:font-character-width for postscript backend that is necessary to solve issue 854.

Fix #854

### CODE REVIEW

**Comment by nsrahmad** [comment]
Can you test if you can run `Drawing Tests` demo? I get:
```lisp
 (clim-demo:demodemo)
> Error: There is no applicable method for the generic function:
>          #<STANDARD-GENERIC-FUNCTION CLIM-BACKEND:FONT-ASCENT #x30200400D79F>
>        when called with arguments:
>          (#<CLX-TTF-PORT :HOST "" :DISPLAY-ID 0 #x3020065CE33D>)
> While executing: #<CCL::STANDARD-KERNEL-METHOD NO-APPLICABLE-METHOD (T)>, in process listener(1).
> Type :GO to continue, :POP to abort, :R for a list of available restarts.
> If continued: Try calling it again
> Type :? for other options.
1 > :B
 (7FBE330116E8) : 0 (FUNCALL #'#<#<CCL::STANDARD-KERNEL-METHOD NO-APPLICABLE-METHOD (T)>> #<STANDARD-GENERIC-FUNCTION CLIM-BACKEND:FONT-ASCENT #x30200400D79F> #<CLX-TTF-PORT :HOST "" :DISPLAY-ID 0 #x3020065CE33D>) 149
 (7FBE33011728) : 1 (FUNCALL #'#<#<STANDARD-METHOD CLIM:TEXT-SIZE (CLIM-CLX:CLX-RENDER-MEDIUM T)>> #<CLIM-CLX:CLX-RENDER-MEDIUM #x3020067A2DBD> "" :TEXT-STYLE NIL :START 0 :END NIL) 229
 (7FBE330117A8) : 2 (%%CHECK-KEYWORDS #(2 #(:END :START :TEXT-STYLE) #<METHOD-FUNCTION CLIM:TEXT-SIZE #>) 17556859527934) 261
 (7FBE33011808) : 3 (FUNCALL #'#<(:INTERNAL #:G207859 (CLIM:COMPOSE-SPACE (CLIM-INTERNALS::DREI-TEXT-FIELD-SUBSTRATE)))> #<CLIM-CLX:CLX-RENDER-MEDIUM #x3020067A2DBD>) 77
 (7FBE33011828) : 4 (%CALL-NEXT-METHOD (NIL #<STANDARD-METHOD CLIM:COMPOSE-SPACE #> . 17556859528014)) 1069
 (7FBE330118A0) : 5 (FUNCALL #'#<#<STANDARD-METHOD CLIM:COMPOSE-SPACE :AROUND (CLIM-INTERNALS::LAYOUT-PROTOCOL-MIXIN)>> (NIL #<STANDARD-METHOD CLIM:COMPOSE-SPACE #> . 17556859528014) :WIDTH #<CLX-44073-DREI-TEXT-FIELD-SUBSTRATE TEXTUAL-DREI-SYNTAX-VIEW #x3020067A681D> :HEIGHT NIL) 141
 (7FBE330118E0) : 6 (%CALL-NEXT-METHOD (NIL #<STANDARD-METHOD CLIM:COMPOSE-SPACE #> . 17556859528014)) 1069
 (7FBE33011958) : 7 (FUNCALL #'#<#<STANDARD-METHOD CLIM:COMPOSE-SPACE :AROUND (CLIM-INTERNALS::SPACE-REQUIREMENT-OPTIONS-MIXIN)>> (NIL #<STANDARD-METHOD CLIM:COMPOSE-SPACE #> . 17556859528014) :WIDTH #<CLX-44073-DREI-TEXT-FIELD-SUBSTRATE TEXTUAL-DREI-SYNTAX-VIEW #x3020067A681D> :HEIGHT NIL) 101
 (7FBE33011998) : 8 (%CALL-NEXT-METHOD (NIL #<STANDARD-METHOD CLIM:COMPOSE-SPACE #> . 17556859528014)) 1069
 (7FBE33011A10) : 9 (%%STANDARD-COMBINED-METHOD-DCODE (#<STANDARD-METHOD CLIM:COMPOSE-SPACE :AROUND #> #<STANDARD-METHOD CLIM:COMPOSE-SPACE :AROUND #> #<STANDARD-METHOD CLIM:COMPOSE-SPACE :AROUND #> #<STANDARD-METHOD CLIM:COMPOSE-SPACE #>) 17556859528014) 773
 (7FBE33011A80) : 10 (%%CHECK-KEYWORDS #(1 #(:HEIGHT :WIDTH) #<Combined-Method CLIM:COMPOSE-SPACE #x30200688210F>) 17556859528025) 261
 (7FBE33011AD8) : 11 (FUNCALL #'#<#<STANDARD-METHOD CLIM-INTERNALS::VERTICALLY-CONTENT-SR** (CLIM-INTERNALS::BOX-LAYOUT-MIXIN T)>> #<CLX-44073-TEXT-FIELD-PANE CLIM-DEMO::BACKEND-FILENAME #x30200679079D> #<CLIM-INTERNALS::BOX-CLIENT #x3020067B332D>) 117
 (7FBE33011B00) : 12 (FUNCALL #'#<#<STANDARD-METHOD CLIM-INTERNALS::BOX-LAYOUT-MIXIN/VERTICALLY-COMPOSE-SPACE (CLIM-INTERNALS::BOX-LAYOUT-MIXIN)>> #<CLX-44073-TEXT-FIELD-PANE CLIM-DEMO::BACKEND-FILENAME #x30200679079D>) 229
 (7FBE33011B98) : 13 (%CALL-NEXT-METHOD (NIL #<STANDARD-METHOD CLIM:COMPOSE-SPACE #> . 17556859528109)) 1069
 (7FBE33011C10) : 14 (FUNCALL #'#<#<STANDARD-METHOD CLIM:COMPOSE-SPACE :AROUND (CLIM-INTERNALS::LAYOUT-PROTOCOL-MIXIN)>> (NIL #<STANDARD-METHOD CLIM:COMPOSE-SPACE #> . 17556859528109) :WIDTH #<CLX-44073-TEXT-FIELD-PANE CLIM-DEMO::BACKEND-FILENAME #x30200679079D> :HEIGHT NIL) 141
 (7FBE33011C50) : 15 (%CALL-NEXT-METHOD (NIL #<STANDARD-METHOD CLIM:COMPOSE-SPACE #> . 17556859528109)) 1069
 (7FBE33011CC8) : 16 (FUNCALL #'#<#<STANDARD-METHOD CLIM:COMPOSE-SPACE :AROUND (CLIM-INTERNALS::SPACE-REQUIREMENT-OPTIONS-MIXIN)>> (NIL #<STANDARD-METHOD CLIM:COMPOSE-SPACE #> . 17556859528109) :WIDTH #<CLX-44073-TEXT-FIELD-PANE CLIM-DEMO::BACKEND-FILENAME #x30200679079D> :HEIGHT NIL) 101
 (7FBE33011D08) : 17 (%%STANDARD-COMBINED-METHOD-DCODE (#<STANDARD-METHOD CLIM:COMPOSE-SPACE :AROUND #> #<STANDARD-METHOD CLIM:COMPOSE-SPACE :AROUND #> #<STANDARD-METHOD CLIM:COMPOSE-SPACE #>) 17556859528109) 773
 (7FBE33011D78) : 18 (%%CHECK-KEYWORDS #(1 #(:HEIGHT :WIDTH) #<Combined-Method CLIM:COMPOSE-SPACE #x302006601DEF>) 17556859528121) 261
 (7FBE33011DD8) : 19 (%CALL-NEXT-METHOD (NIL #<STANDARD-METHOD CLIM:COMPOSE-SPACE #> . 17556859528204)) 1069
 (7FBE33011E50) : 20 (FUNCALL #'#<#<STANDARD-METHOD CLIM:COMPOSE-SPACE (CLIM:SPACING-PANE)>> (NIL #<STANDARD-METHOD CLIM:COMPOSE-SPACE #> . 17556859528204) :WIDTH #<CLX-44073-LOWERED-PANE NIL #x3020067B306D> :HEIGHT NIL) 101
 (7FBE33011E90) : 21 (%CALL-NEXT-METHOD (NIL #<STANDARD-METHOD CLIM:COMPOSE-SPACE #> . 17556859528204)) 1069
 (7FBE33011F08) : 22 (FUNCALL #'#<#<STANDARD-METHOD CLIM:COMPOSE-SPACE :AROUND (CLIM-INTERNALS::LAYOUT-PROTOCOL-MIXIN)>> (NIL #<STANDARD-METHOD CLIM:COMPOSE-SPACE #> . 17556859528204) :WIDTH #<CLX-44073-LOWERED-PANE NIL #x3020067B306D> :HEIGHT NIL) 141
 (7FBE33011F48) : 23 (%CALL-NEXT-METHOD (NIL #<STANDARD-METHOD CLIM:COMPOSE-SPACE #> . 17556859528204)) 1069
 (7FBE33011FC0) : 24 (FUNCALL #'#<#<STANDARD-METHOD CLIM:COMPOSE-SPACE :AROUND (CLIM-INTERNALS::SPACE-REQUIREMENT-OPTIONS-MIXIN)>> (NIL #<STANDARD-METHOD CLIM:COMPOSE-SPACE #> . 17556859528204) :WIDTH #<CLX-44073-LOWERED-PANE NIL #x3020067B306D> :HEIGHT NIL) 101
 (7FBE33012000) : 25 (%%STANDARD-COMBINED-METHOD-DCODE (#<STANDARD-METHOD CLIM:COMPOSE-SPACE :AROUND #> #<STANDARD-METHOD CLIM:COMPOSE-SPACE :AROUND #> #<STANDARD-METHOD CLIM:COMPOSE-SPACE #> #<STANDARD-METHOD CLIM:COMPOSE-SPACE #>) 17556859528204) 773
 (7FBE33012070) : 26 (%%CHECK-KEYWORDS #(1 #(:HEIGHT :WIDTH) #<Combined-Method CLIM:COMPOSE-SPACE #x3020067D353F>) 17556859528215) 261
 (7FBE330120C8) : 27 (FUNCALL #'#<#<STANDARD-METHOD CLIM-INTERNALS::HORIZONTALLY-CONTENT-SR** (CLIM-INTERNALS::BOX-LAYOUT-MIXIN T)>> #<CLX-44073-HRACK-PANE NIL #x3020067B2D5D> #<CLIM-INTERNALS::BOX-CLIENT #x3020067B2B0D>) 117
 (7FBE330120F0) : 28 (FUNCALL #'#<#<STANDARD-METHOD CLIM-INTERNALS::BOX-LAYOUT-MIXIN/HORIZONTALLY-COMPOSE-SPACE (CLIM-INTERNALS::BOX-LAYOUT-MIXIN)>> #<CLX-44073-HRACK-PANE NIL #x3020067B2D5D>) 229
```

**Comment by admich** [comment]
Yes, I also have the error. Sorry I didn't try drawing-test demo before. I will investigate the causes.

**Comment by admich** [comment]
@nsrahmad I have found the problem. Now it works. It was a stupid error I typed (find-port) instead of (find-font) in a method definition.

File: Backends/PostScript/font/font.lisp [hunk 1]
```diff
@@ -300,3 +299,12 @@
       (define-font-metrics dict-name ascent descent angle char-infos font-name))
     (setf (text-style-mapping port text-style) font-name)
     text-style))
+
+(defmethod climb:font-character-width :around ((font postscript-font-name) character)
```
Comment (nsrahmad) [inline Backends/PostScript/font/font.lisp]: This probably doesn't need to be an `:around` method.

Comment (admich) [reply Backends/PostScript/font/font.lisp]: you are right, follo a force push

**Pull request merged**